packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -3182,7 +3194,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           this.raise(
             method.start,
             TSErrors.AbstractMethodHasImplementation,
-            key.type === "Identifier"
+            !method.computed
```

Comment:
A non-computed property key can be string literal or numeric literal, which do not have `.name` property.

```ts
abstract class Foo {
  abstract 1() {}
  abstract "2"() {}
}
```